lib/python/rose/macro.py

```diff
@@ -1276,23 +1296,97 @@ def _report_error(exception=None, text=""):
     )
 
 
+def get_apps_from_dir(conf_dir):
+    """Returns a list of apps with respect to the provided directory.
+    If the directory is an app return that app.
+    If the directory contains apps return the paths of those apps.
+    If the directory is a suite return the paths of its apps."""
+    config_file_path = os.path.join(conf_dir,
+                                    rose.SUB_CONFIG_NAME)
+    if (os.path.exists(config_file_path) or os.path.isfile(config_file_path)):
+        # we are in an app
+        return DIR_TYPE_APP, [conf_dir]
+    else:
+        # are we in an apps folder?
+        apps = []
+        for _, subdirs, _ in os.walk(conf_dir):
+            for dir_ in subdirs:
+                config_file_path = os.path.join(
+                    conf_dir,
+                    dir_,
+                    rose.SUB_CONFIG_NAME
+                )
+                if (os.path.exists(config_file_path) and
+                        os.path.isfile(config_file_path)):
+                    apps.append(os.path.join(conf_dir, dir_))
+            break
+        if apps != []:
+            return DIR_TYPE_APP_DIR, apps
+
+        # are we in a suite directory
+        config_file_path = os.path.join(conf_dir, rose.TOP_CONFIG_NAME)
+        if (os.path.exists(config_file_path) and
+                os.path.isfile(config_file_path)):
+            for _, subdirs, _ in os.walk(os.path.join(conf_dir,
```

Comment:
I know it sounds weird, but there can be apps with `rose-app.conf` files a couple of levels deep within their own directory structure! It's probably easier to do the same as we do with optional configuration files above (`glob.glob`) and only look for SUB_CONFIG_DIR + "/*/" + SUB_CONFIG_NAME.